File: calculator.py
class SimpleCalculator:
    def add(self, a, b):
        return a + b

    def subtract(self, a, b):
        return a - b

    def multiply(self, a, b):
        return a * b

    def divide(self, a, b):
        if b == 0:
            raise ValueError("Cannot divide by zero")
        return a / b

    def factorial(self, n):
        if n == 0:
            return 1
        else:
            return n * self.factorial(n - 1)

def main():
    calculator = SimpleCalculator()
    operation = input("Enter operation (add, subtract, multiply, divide, factorial): ").strip().lower()
    if operation == "factorial":
        n = int(input("Enter a number: "))
        print(f"Factorial: {calculator.factorial(n)}")
    else:
        a = float(input("Enter first number: "))
        b = float(input("Enter second number: "))
        if operation == "add":
            print(f"Addition: {calculator.add(a, b)}")
        elif operation == "subtract":
            print(f"Subtraction: {calculator.subtract(a, b)}")
        elif operation == "multiply":
            print(f"Multiplication: {calculator.multiply(a, b)}")
        elif operation == "divide":
            print(f"Division: {calculator.divide(a, b)}")
        else:
            print("Invalid operation")

if __name__ == "__main__":
    main()

File: calculator.js
class SimpleCalculator {
    add(a, b) {
        return a + b;
    }

    subtract(a, b) {
        return a - b;
    }

    multiply(a, b) {
        return a * b;
    }

    divide(a, b) {
        if (b === 0) {
            throw new Error("Cannot divide by zero");
        }
        return a / b;
    }

    factorial(n) {
        if (n === 0) {
            return 1;
        } else {
            return n * this.factorial(n - 1);
        }
    }
}

function main() {
    const calculator = new SimpleCalculator();
    const readline = require('readline').createInterface({
        input: process.stdin,
        output: process.stdout
    });

    readline.question("Enter operation (add, subtract, multiply, divide, factorial): ", operation => {
        operation = operation.trim().toLowerCase();
        if (operation === "factorial") {
            readline.question("Enter a number: ", n => {
                console.log(`Factorial: ${calculator.factorial(parseInt(n))}`);
                readline.close();
            });
        } else {
            readline.question("Enter first number: ", a => {
                readline.question("Enter second number: ", b => {
                    a = parseFloat(a);
                    b = parseFloat(b);
                    switch (operation) {
                        case "add":
                            console.log(`Addition: ${calculator.add(a, b)}`);
                            break;
                        case "subtract":
                            console.log(`Subtraction: ${calculator.subtract(a, b)}`);
                            break;
                        case "multiply":
                            console.log(`Multiplication: ${calculator.multiply(a, b)}`);
                            break;
                        case "divide":
                            try {
                                console.log(`Division: ${calculator.divide(a, b)}`);
                            } catch (error) {
                                console.log(error.message);
                            }
                            break;
                        default:
                            console.log("Invalid operation");
                    }
                    readline.close();
                });
            });
        }
    });
}

main();